app/carbonzipper/app.go

```diff
@@ -294,3 +178,114 @@ func initBackends(config cfg.Zipper, logger *zap.Logger) ([]backend.Backend, err
 
 	return backends, nil
 }
+
+func initGraphite(app *App) {
+	// register our metrics with graphite
+	graphite := g2g.NewGraphite(app.config.Graphite.Host, app.config.Graphite.Interval, 10*time.Second)
+
+	/* #nosec */
+	hostname, _ := os.Hostname()
+	hostname = strings.Replace(hostname, ".", "_", -1)
+
+	prefix := app.config.Graphite.Prefix
+
+	pattern := app.config.Graphite.Pattern
+	pattern = strings.Replace(pattern, "{prefix}", prefix, -1)
+	pattern = strings.Replace(pattern, "{fqdn}", hostname, -1)
+
+	graphite.Register(fmt.Sprintf("%s.requests", pattern), Metrics.Requests)
+	graphite.Register(fmt.Sprintf("%s.responses", pattern), Metrics.Responses)
+	graphite.Register(fmt.Sprintf("%s.errors", pattern), Metrics.Errors)
+
+	graphite.Register(fmt.Sprintf("%s.find_requests", pattern), Metrics.FindRequests)
+	graphite.Register(fmt.Sprintf("%s.find_errors", pattern), Metrics.FindErrors)
+
+	graphite.Register(fmt.Sprintf("%s.render_requests", pattern), Metrics.RenderRequests)
+	graphite.Register(fmt.Sprintf("%s.render_errors", pattern), Metrics.RenderErrors)
+
+	graphite.Register(fmt.Sprintf("%s.info_requests", pattern), Metrics.InfoRequests)
+	graphite.Register(fmt.Sprintf("%s.info_errors", pattern), Metrics.InfoErrors)
+
+	graphite.Register(fmt.Sprintf("%s.timeouts", pattern), Metrics.Timeouts)
+
+	for i := 0; i <= app.config.Buckets; i++ {
+		graphite.Register(fmt.Sprintf("%s.requests_in_%dms_to_%dms", pattern, i*100, (i+1)*100), bucketEntry(i))
+		lower, upper := util.Bounds(i)
+		graphite.Register(fmt.Sprintf("%s.exp.requests_in_%05dms_to_%05dms", pattern, lower, upper), expBucketEntry(i))
+	}
+
+	graphite.Register(fmt.Sprintf("%s.cache_size", pattern), Metrics.CacheSize)
+	graphite.Register(fmt.Sprintf("%s.cache_items", pattern), Metrics.CacheItems)
+
+	graphite.Register(fmt.Sprintf("%s.cache_hits", pattern), Metrics.CacheHits)
+	graphite.Register(fmt.Sprintf("%s.cache_misses", pattern), Metrics.CacheMisses)
+
+	go mstats.Start(app.config.Graphite.Interval)
+
+	graphite.Register(fmt.Sprintf("%s.goroutines", pattern), Metrics.Goroutines)
+	graphite.Register(fmt.Sprintf("%s.uptime", pattern), Metrics.Uptime)
+	graphite.Register(fmt.Sprintf("%s.alloc", pattern), &mstats.Alloc)
+	graphite.Register(fmt.Sprintf("%s.total_alloc", pattern), &mstats.TotalAlloc)
+	graphite.Register(fmt.Sprintf("%s.num_gc", pattern), &mstats.NumGC)
+	graphite.Register(fmt.Sprintf("%s.pause_ns", pattern), &mstats.PauseNS)
+}
+
+func metricsServer(app *App, logger *zap.Logger) {
+	prometheus.MustRegister(app.prometheusMetrics.Requests)
```

Comment:
This can also be extracted and moved to a separate function.

Reply:
Absolutely. This one is in progress.